FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift

```diff
@@ -264,15 +264,20 @@ import Foundation
                                                   enrollmentInfo: startMFARequestInfo,
                                                   requestConfiguration: auth.requestConfiguration)
           let response = try await auth.backend.call(with: request)
-          return response.phoneSessionInfo?.sessionInfo
+          guard let sessionInfo = response.phoneSessionInfo?.sessionInfo else {
+            throw AuthErrorUtils.error(code: .missingMultiFactorInfo)
+          }
+          return sessionInfo
```

Comment:
Here and below, we need to unwrap the underlying session info which is current an optional:

https://github.com/firebase/firebase-ios-sdk/blob/380a8618823f1fa7440cb30059025710ae1d7e5e/FirebaseAuth/Sources/Swift/Backend/RPC/Proto/Phone/AuthProtoStartMFAPhoneResponseInfo.swift#L17-L23

Throwing an error is new behavior here. I think this may be ok as long there aren't success cases where the sessionInfo _should be_ `nil`?